mini-projects/00-jq-rs/src/printers.rs

```diff
@@ -0,0 +1,243 @@
+use ansi_term::Colour;
+use serde::Serialize;
+use serde_json::Value;
+
+pub struct Formatter {
+    pub color_output: bool,
+    pub indent: usize,
+    pub compact: bool,
+    pub sort_keys: bool,
```

Comment:
Rather than having multiple parameters within this one formatting struct, you could have split this into more dedicate formatters (like a ColorFormatter) to avoid excessive if statements in the `print_value` function